pdns/ws-recursor.cc

```diff
@@ -437,13 +437,14 @@ static void prometheusMetrics(HttpRequest *req, HttpResponse *resp) {
 
     std::ostringstream output;
     typedef map <string, string> varmap_t;
-    varmap_t varmap = getAllStatsMap(
-            StatComponent::API); // Argument controls blacklisting of any stats. So stats-api-blacklist will be used to block returned stats.
+    
+    // Argument controls blacklisting of any stats. So
+    // stats-api-blacklist will be used to block returned stats.
+    // Second arg tells to use the prometheus names.
+    varmap_t varmap = getAllStatsMap(StatComponent::API, true);
     for (const auto &tup :  varmap) {
         std::string metricName = tup.first;
-
-        // Prometheus suggest using '_' instead of '-'
-        std::string prometheusMetricName = "pdns_recursor_" + boost::replace_all_copy(metricName, "-", "_");
+        std::string prometheusMetricName = "pdns_recursor_" + metricName;
```

Comment:
If the Prometheus was explicitly set, it must not be prefixed with `pdns_recursor_`.

Reply:
Right. I'll move the prepending of the prefix to the proper place.